util/validate.go

```diff
@@ -0,0 +1,26 @@
+package util
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/prometheus/common/model"
+)
+
+var validLabelRE = regexp.MustCompile(`^[a-zA-Z_][a-zA-Z0-9_]*$`)
+
+// ValidateSample returns an err if the sample is invalid
+func ValidateSample(s *model.Sample) error {
```

Comment:
Danger of this approach is that we need to remember to call `ValidateSample` before handling any incoming samples. A more robust solution would be to return a validated sample that has different type to the not-yet-validated sample.

However, not necessary for this PR, and maybe not ever.